tests/rptest/scale_tests/extreme_recovery_test.py

```diff
@@ -0,0 +1,208 @@
+                p = FranzGoVerifiableProducer(self.ctx, self.redpanda,
+                                              topic.name, self.MSG_SIZE,
+                                              self._msg_per_topic)
+                p.start()
+                producers.append(p)
+
+        # wait for completion
+        for i, p in enumerate(producers):
+            self.redpanda.logger.info(
+                f"gen baseline: wait for producer {i+1}.. ")
+            p.wait()
+        t1 = time()
+        for p in producers:
+            p.free()
+        delta_t = t1 - t0
+        gib = self._total_bytes / 2**30
+        mib = self._total_bytes / 2**20
+        self.logger.info(
+            f"Produced {gib:0.3f} GiB in {delta_t:0.3f}s, "
+            f"{mib/delta_t:0.3f} MiB/s.")
+
+    def validate_cluster(self, baseline, restored):
+        """Validate all data after recovery."""
+        self.logger.info(
+            f"vallidate_cluster base:restored: {baseline}:{restored}")
+
+        self._validate_partition_last_offset()
+        expected_topics = [
+            topic.name for topic in self.expected_recovered_topics
+        ]
+
+        # consume and validate all data
+        t0 = time()
+        consumers: list[FranzGoVerifiableSeqConsumer] = list()
+        for topic in self.topics:
+            self.logger.info(
+                f'validate_cluster: starting consumer for "{topic}"')
+            c = FranzGoVerifiableSeqConsumer(self.ctx, self.redpanda, topic,
+                                             self.MSG_SIZE)
+            c.start()
+
+        for c in consumers:
+            self.logger.info(
+                f'validate_cluster: waiting on consumer for "{c._topic}"')
+            c.shutdown()
+            c.wait()
+            assert c.consumer_status.invalid_reads == 0, "No invalid reads expected."
+            assert c.consumer_status.valid_reads == self._msg_per_topic
+
+        delta_t = time() - t0
+        self.logger.info(
+            f'validate_cluster: consumed {self._total_bytes/2**20}MB in {delta_t} sec;'
+            f' {(self._total_bytes/delta_t)/2**20} MB/s.')
+
+    @property
+    def second_restart_needed(self):
+        """Return True if the restart is needed afer first validation steps"""
+        # XXX TODO
+        return False
+
+    def after_restart_validation(self):
+        """Check that topic is writable after restart"""
+        # XXX TODO
+
+
+# "It's like, totally extreme, bro."
+# Try 1: hack topic recovery test into being a scale test.
+class ExtremeRecoveryTest(TopicRecoveryTest):
+    def __init__(self, test_context: TestContext):
+        extra_rp_conf = {"blah": "bloo"}
+        super(ExtremeRecoveryTest,
+              self).__init__(test_context,
+                             num_brokers=RecoveryScale.NODE_COUNT)
+        self.initial_cleanup_needed = True
+        self.ctx = test_context
+
+    def tearDown(self):
+        super(ExtremeRecoveryTest, self).tearDown()
```

Comment:
This isn't needed unless you intend to add something specific for this class.

Reply:
Yeah, so far it is just a reminder that "any shutdown code you need goes here".